qiskit/primitives/utils.py

```diff
@@ -54,23 +54,16 @@ def init_observable(observable: BaseOperator | str) -> SparsePauliOp:
     Returns:
         The observable as :class:`~qiskit.quantum_info.SparsePauliOp`.
 
+    Raises:
+        QiskitError: when observable type cannot be converted to SparsePauliOp.
     """
 
     if isinstance(observable, SparsePauliOp):
         return observable
     elif isinstance(observable, BaseOperator) and not isinstance(observable, BasePauli):
         return SparsePauliOp.from_operator(observable)
     else:
-        if isinstance(observable, PauliList):
-            warnings.warn(
-                "Implicit conversion from a PauliList to a SparsePauliOp with coeffs=1 in"
-                " estimator observable arguments is deprecated as of Qiskit 0.46 and will be"
-                " in Qiskit 1.0. You should explicitly convert to a SparsePauli op using"
-                " SparsePauliOp(pauli_list) to avoid this warning.",
-                DeprecationWarning,
-                stacklevel=2,
-            )
-        return SparsePauliOp(observable)
+        raise QiskitError(f"observable type not supported: {type(observable)}")
```

Comment:
This is a behavioural change compared to the previous state - the warning only affected `PauliList`, but the new action forbids anything that hit this `else` path.